### PR TITLE
Make tag search arguments arrays grow geometrically.

### PR DESCRIPTION
**Pull request by postwait**
Grow by x2 instread of +1.

### CODE REVIEW

File: src/noit_metric_tag_search.c [hunk 1]
```diff
@@ -580,19 +583,20 @@ typedef struct noit_metric_tag_search_ast_t {
...
+noit_metric_tag_search_resize_args(noit_metric_tag_search_ast_t *node, int new_size) {
   assert(node->operation == OP_NOT_ARGS ||
          node->operation == OP_OR_ARGS ||
          node->operation == OP_AND_ARGS);
-  if(node->contents.args.cnt == cnt) return;
-  assert(node->contents.args.cnt <= cnt);
-  noit_metric_tag_search_ast_t **new_nodes = calloc(cnt, sizeof(*new_nodes));
+  if(new_size == 0) new_size = DEFAULT_CHILDREN_ALLOC;
+  if(node->contents.args.nallocd >= new_size) return;
+  assert(node->contents.args.nallocd <= new_size);
```
Comment (jonInfinity) [inline src/noit_metric_tag_search.c:592]: This isn't required, line 591 would have already returned

File: src/noit_metric_tag_search.c [hunk 2]
```diff
@@ -666,9 +670,13 @@ void
...
   assert(node->operation != OP_NOT_ARGS || idx == 0);
-  if(idx == node->contents.args.cnt) noit_metric_tag_search_resize_args(node, idx+1);
-  assert(node->contents.args.cnt > idx);
+  noit_metric_tag_search_resize_args(node, node->contents.args.nallocd*2);
+  assert(node->contents.args.nallocd > idx);
+  noit_metric_tag_search_ast_t *prev = node->contents.args.node[idx];
   node->contents.args.node[idx] = r;
+  assert(node->contents.args.cnt >= idx); // can't insert with gaps.
+  if(node->contents.args.cnt == idx) node->contents.args.cnt++;
+  else noit_metric_tag_search_free(prev);
```
Comment (jonInfinity) [inline src/noit_metric_tag_search.c:678]: What happens to prev if node->contents.args.cnt == idx ? It looks like it will leak memory

Comment (postwait) [reply src/noit_metric_tag_search.c:678]: If cnt == idx we're appending to the end of an existing list (and not replacing an existing item).  The only idx's that have items in the are less than cnt.

**Review by jonInfinity — APPROVED** [review]
LGTM

**Pull request merged**